tests_e2e/scenarios/testsuites/agent_junit.py

```diff
@@ -0,0 +1,50 @@
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from dataclasses import dataclass
+from typing import Type
+
+from dataclasses_json import dataclass_json
+
+from lisa.notifiers.junit import JUnit
+from lisa import schema
+from lisa.messages import (
+    MessageBase,
+    TestResultMessage,
+)
+
+
+@dataclass_json()
+@dataclass
+class AgentJUnitSchema(schema.Notifier):
+    path: str = "agent.junit.xml"
+
+
+class AgentJUnit(JUnit):
+    @classmethod
+    def type_name(cls) -> str:
+        return "agent.junit"
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return AgentJUnitSchema
+
+    def _received_message(self, message: MessageBase) -> None:
+        if isinstance(message, TestResultMessage):
+            distro = message.information.get('distro_version')
+            if distro is not None:
+                message.name = distro
+        super()._received_message(message)
```

Comment:
We replace the test name (which is always "main" in our test runs) with the distro name